dot/types/inherents.go

```diff
@@ -43,24 +62,20 @@ func (d *InherentsData) String() string {
 	return str
 }
 
-// SetInt64Inherent set an inherent of type uint64
-func (d *InherentsData) SetInt64Inherent(key []byte, data uint64) error {
-	if len(key) != 8 {
-		return errors.New("inherent key must be 8 bytes")
+// SetInherent sets a inherent.
+func (d *InherentsData) SetInherent(inherentIdentifier InherentIdentifier, value interface{}) error {
```

Comment:
supernit we could use `any` now that we use go 1.18

```suggestion
func (d *InherentsData) SetInherent(inherentIdentifier InherentIdentifier, value any) error {
```